components/core/tests/test-NetworkReader.cpp

```diff
@@ -97,6 +100,23 @@ auto assert_curl_error_code(CURLcode expected, clp::NetworkReader const& reader)
     WARN(message_to_log);
     return false;
 }
+
+auto test_illegal_header(std::string const& header_name, std::string const& header_value) -> bool {
```

Comment:
Looking at it again, I feel like we should create a stand-alone test case for illegal header testing, and use `GENERATE` to create a list of illegal kv pairs for testing instead of having this helper function. But please read the comments above as they're the general rule for adding a helper function.